src/ServiceBus.AttachmentPlugin/TokenGenerator.cs

```diff
@@ -0,0 +1,27 @@
+﻿namespace ServiceBus.AttachmentPlugin
+{
+    using Microsoft.WindowsAzure.Storage.Blob;
+    using System;
+
+    static class TokenGenerator
+    {
+        internal static string GetBlobSasUri(CloudBlockBlob blob, TimeSpan timeSpan)
+        {
+            //Set the expiry time and permissions for the blob.
+            //In this case the start time is specified as a few minutes in the past, to mitigate clock skew.
+            //The shared access signature will be valid immediately.
+            var sasConstraints = new SharedAccessBlobPolicy
+            {
+                SharedAccessStartTime = DateTime.UtcNow.AddMinutes(-5),
+                SharedAccessExpiryTime = DateTime.UtcNow.Add(timeSpan),
+                Permissions = SharedAccessBlobPermissions.Read
```

Comment:
Read and Delete so that clean-up could be performed when needed.

Reply:
I might not fully agree that that we should have delete. There is a scenario where a message is put on a topic and reaches several destinations. What happens if one deletes the message before others can read it?
Im changing according to your comment now though.

Reply:
In the described scenario I don't see an issue. A topic with multiple subscription would replicate the message to each subscription. Each subscriber would delete its copy, not a shared message. So should be all safe and sound.